library/math/complex.lisp

```diff
@@ -91,39 +99,44 @@ below.")
   ;;   (define (general/ a b) (complex-divide a b)))
 
   (define-instance ((Complex :a) (Reciprocable :a) => Reciprocable (Complex :a))
+    (inline)
     (define (reciprocal x)
-      (let a = (real-part x))
-      (let b = (imag-part x))
-      (let divisor = (reciprocal (square-magnitude x)))
-      ;; z^-1 = z*/|z|^2
-      (complex (* a divisor) (negate (* b divisor))))
+      (complex-reciprocal x))
+    (inline)
     (define (/ a b)
       (complex-divide a b)))
 
   (define-instance (Complex :a => Complex (Complex :a))
+    (inline)
     (define (complex a b)
       (%Complex a b))
+    (inline)
     (define (real-part a)
       (match a
         ((%Complex a _) a)))
+    (inline)
     (define (imag-part a)
       (match a
         ((%Complex _ b) b))))
 
   ;; Below are specializable functions, as class methods cannot be specialized
   ;; This allows us to call out to faster lisp functions for doing arithmetic.
   ;; These will only be called from monomorphized forms.
+  (inline)
```

Comment:
i wouldn't inline these functions, because for the average data type, we won't benefit

we will from inlining specializations though